### ISSUE TITLE
Client options

### ISSUE THREAD

**Issue by reportaman** [tmux/tmux#2809]
### Issue description

As of now it is possible to set global, server, session, window, and pane options. But it's not possible to set options for the current client (afaik). Having client options (local to client tty) will make it possible to code wakatime-like functionality to log time one spends in a pane while also being able to switch directly between different clients. When I had requested client-focus-out event, I did not consider the case where the focus is sent directly to another client. As client-focus-out, pane-focus-in are not strictly ordered, it requires post processing to figure out correct duration to log. Using client-scoped variables will avoid a race, and one would be able to compute duration spent in a pane without having to do post processing because of unorderedness of events when focus is moving from one client to another. With no need for post processing, perhaps a cli application can be written to show a live text-graph dashboard of user activity in one of the panes.

> If there is another way of setting tty-local options, that might work too?

### Required information

Please provide the following information:

* tmux version (`tmux -V`): tmux next-3.3.
* Platform (`uname -sp`) Darwin arm.
* $TERM inside and outside of tmux (`echo $TERM`): xterm-256color outside, tmux-256color inside.

**Comment by nicm** [comment]
What do you mean pane focus in/out events are not ordered?

I don't think client options are worth it, you will need to do it using global options with the name of the client in, or a file.

**Comment by reportaman** [comment]
@nicm Using global options with name of the client could work if there is a way to figure which client/s detached, so one is able to clear the global variables which are prefixed with client name. That way one can unset the client variables, and make sure there is no steady increase in global memory.

**Comment by nicm** [comment]
Can you try this which moves the setup of the format variables earlier and adds `hook_client`? [notify.diff.txt](https://github.com/tmux/tmux/files/6966962/notify.diff.txt)

**Comment by reportaman** [comment]
@nicm Will test later tonight, and report. But before that it would be good to understand what you mean by 

> ...which moves the setup of the format variables earlier...

Does that mean all the hook options like hook_pane, hook_session, are valid & available for all hooks now?

**Comment by reportaman** [comment]
Ok, I printed hook_client, hook_session, hook_window, and hook_pane for events clientfocusin, clientfocusout, clientsessionchanged, panefocusin, panefocusout. Tried it with 2-3 clients side by side, and with either same or different sessions attached.

1. clientsessionchanged, clientfocusin, clientfocusout print valid hook_client tty-name.
2. panefocusin, panefocusout print valid hook_pane pane-id.

For all these events, hook_session, hook_window are empty as was the case in past & I assume we continue using tmux display-message to derive correct session, and window.

Works well I think!
